dependency/vault_read.go

```diff
@@ -82,7 +83,7 @@ func (d *VaultReadQuery) Fetch(clients *ClientSet, opts *QueryOptions,
 	if !vaultSecretRenewable(d.secret) {
 		dur := leaseCheckWait(d.secret)
 		log.Printf("[TRACE] %s: non-renewable secret, set sleep for %s", d, dur)
-		d.sleepCh = time.After(dur)
+		d.sleepCh <- time.After(dur)
```

Comment:
hm 🤔 this chan(chan) is confusing me. I'm wondering if this can be done in a simpler way?
I'm sure I'm missing something -- but why can't this line just be `time.Sleep(dur)` and get rid of all the sleep channel stuff? `time.After` will block in the same way a `time.Sleep` would.. I guess I don't understand the use of the channel here. Could you explain your thinking to me?

Reply:
The main point is that the sleep needs to be saved until next time the function is called. The function is called repeatedly by the polling code and is responsible for its own retry timing (sleeping). The first time it is run it needs to return immediately, then after that it needs to have a delay. The sleep/delay is checked for and done at the top of that function (~ line 64).

It could be simplified into `chan time.Duration` though... the channel of channels is not necessary. That would simplify it a bit. It would change where it is used like...

From..
```
	select {
	case sleep := <-d.sleepCh:
		<-sleep
	default:
	}
```
To.. (assuming d.sleepCh was `chan time.Duration`)..
``` 
	select {
	case dur := <-d.sleepCh:
		time.Sleep(dur)
	default:
	}
```

Reply:
I see! Thanks for explaining. Oh man I like that second way (w/ `time.Sleep(dur)`) so much better, it's a lot clearer to me. Wdyt?

Reply:
I agree, it is cleaner. I think I was just stuck in the time.After rut and didn't really consider alternatives.

Reply:
Just force-pushed that change (along with a re-wording of the commit message).